Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -7737,7 +7737,7 @@ HTMLMediaElement::SleepType HTMLMediaElement::shouldDisableSleep() const
 #if !PLATFORM(COCOA) && !PLATFORM(GTK) && !PLATFORM(WPE)
     return SleepType::None;
 #endif
-    if (m_sentEndEvent || !m_player || m_player->paused() || loop())
+    if (m_sentEndEvent || !m_player || !m_player->timeIsProgressing() || loop())
```

Comment:
It seems we should be able to write a test that checks shouldDisableSleep via internals API.
Can we do that for the various players if this is not already the case?

Reply:
we have: already `media/media-source/media-source-stalled-holds-sleep-assertion.html` it's this test that started to fail after PR#25565

Reply:
OK, I see tests for MSE and AVFoundation. We might be missing tests for webm and media stream.
I filed https://bugs.webkit.org/show_bug.cgi?id=270625.